File: Selenium/Activity5.py
from selenium import webdriver
from selenium.webdriver.common.by import By

with webdriver.Firefox() as driver:

    driver.get("https://training-support.net/webelements/dynamic-controls")

    print("Page title:", driver.title)

    is_visible = len(driver.find_elements(By.ID, "checkbox")) > 0
    print("Checkbox visible before click:", is_visible)

    driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']").click()

    is_visible = len(driver.find_elements(By.ID, "checkbox")) > 0
    print("Checkbox visible after removal:", is_visible)

    driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']").click()

    is_visible = len(driver.find_elements(By.ID, "checkbox")) > 0
    print("Checkbox visible after adding back:", is_visible)

File: Selenium/Activity5.java
package examples;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity5 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/dynamic-controls");

        System.out.println("Page title: " + driver.getTitle());

        boolean isVisible = driver.findElements(By.id("checkbox")).size() > 0;
        System.out.println("Checkbox visible before click: " + isVisible);

        driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();

        isVisible = driver.findElements(By.id("checkbox")).size() > 0;
        System.out.println("Checkbox visible after removal: " + isVisible);

        driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();

        isVisible = driver.findElements(By.id("checkbox")).size() > 0;
        System.out.println("Checkbox visible after adding back: " + isVisible);

        driver.quit();
    }
}
